tests/appservice/test_api.py

```diff
@@ -76,15 +76,15 @@ async def get_json(
             headers: Mapping[Union[str, bytes], Sequence[Union[str, bytes]]],
```

Comment:
I'm guessing this is meant to match the signature of `get_json` on the http client in Synapse that's being mocked out? If so, it seems to me that the least surprising thing would be to make the mock accept both strs and bytes(?)

Reply:
I think you mean https://github.com/matrix-org/synapse/blob/d35bed8369514fe727b4fe1afb68f48cc8b2655a/synapse/http/client.py#L532-L537 right?

That one uses "RawHeaders". Which over N corners comes basically to a type close to this one. Since twisted accepts both bytes and strings as well (in fact, it seems to internally encode it as bytes anyway. https://github.com/twisted/twisted/blob/a80a551234651918961d41c39e3706b2162cb226/src/twisted/web/http_headers.py#L86-L88 )

Reply:
Though long story short: The mock probably should be updated to be the same types right? Should I do this in the PR or should that happen elsewhere?

Reply:
Yep, that's the one.

> The mock probably should be updated to be the same types right?

Ideally yes, but I think if this passes CI then we should land it as-is and leave the types for another PR.

Related: #8812, #14301, #14303. In particular it looks like I didn't think about the tests when I wrote #14303.